src/core/src/engine.rs

```diff
@@ -470,36 +473,37 @@ fn validtor_has_claimed_block_limit( service: &mut Poet2Service ) -> bool {
 
 
 //c-test
-//      fn validator_is_claiming_too_early(&mut self , block_id : BlockId ,block_number : i32 ,validator_registry_view:ValidatorRegistryView , block_store:Blockstore)
-fn validator_is_claiming_too_early( service: &mut Poet2Service )->bool
+fn validator_is_claiming_too_early(block: Block ,service: &mut Poet2Service) -> bool
 {
 
-    let number_of_validators = 32;
-    let total_block_claim_count = 33 ;
-    let commit_block_block_num = 2;
-    let block_number = 5 ;
-//  number_of_validators = (validator_registry_view.get_validators()).len();  //stubbed function
+    let number_of_validators = 3;
+    //    number_of_validators = (validator_registry_view.get_validators()).len();  //stubbed function
+    let total_block_claim_count = block.block_num - 1;
+    let commit_block_block_num = 0;
+    //    let commit_block = block_store.get_block_by_transaction_id(validator_info.transaction_id)
+    let block_number = block.block_num;
+
     let block_claim_delay_from_settings = service.get_setting_from_head(
         String::from("sawtooth.poet.block_claim_delay"));
 
     let key_block_claim_delay = block_claim_delay_from_settings.parse::<i32>().unwrap();
     let block_claim_delay = cmp::min(key_block_claim_delay, number_of_validators - 1);
 
-    if total_block_claim_count <= block_claim_delay
+    if total_block_claim_count <= block_claim_delay as u64
     {
         return false;
     }
     // need to use get_block from service expecting block_id to have been stored
     // along with validator info in the Poet 2 module
 
-//  let commit_block = block_store.get_block_by_transaction_id(validator_info.transaction_id)  //
-
-    let blocks_claimed_since_registration = block_number - commit_block_block_num - 1 ;
+    let blocks_claimed_since_registration  = block_number - commit_block_block_num - 1 ;
 
-    if block_claim_delay > blocks_claimed_since_registration
+    if block_claim_delay > blocks_claimed_since_registration as i32
```

Comment:
Most of the the numerical variables here are counts which cannot be less than 0. So prefer u32/u64 over i32/i64.